text/1162449/undirected_ring_stages_with_feedback.tex

```diff
@@ -0,0 +1,99 @@
+\documentclass{article}
+
+\usepackage[english]{babel}
+\usepackage{filecontents}
+\usepackage[letterpaper,top=2cm,bottom=2cm,left=3cm,right=3cm,marginparwidth=1.75cm]{geometry}
+
+\usepackage{amsmath}
+\usepackage{csquotes}
+\usepackage{graphicx}
+\usepackage{amsthm}
+\usepackage[colorlinks=true, allcolors=blue]{hyperref}
+\begin{filecontents}{\jobname.bib}
+@article{ROTEM1987575,
+title = {Analysis of a distributed algorithm for extrema finding in a ring},
+journal = {Journal of Parallel and Distributed Computing},
+volume = {4},
+number = {6},
+pages = {575-591},
+year = {1987},
+issn = {0743-7315},
+doi = {https://doi.org/10.1016/0743-7315(87)90031-1},
+url = {https://www.sciencedirect.com/science/article/pii/0743731587900311},
+author = {D. Rotem and E. Korach and N. Santoro},
+abstract = {A new and more detailed analysis of the unidirectional algorithm of Chang and Roberts for distributed extrema finding on a ring is presented. This analysis shows that this simple algorithm, which is known to be average case optimal, compares very favorably with all the other known algorithms as it requires O(n log n) messages with probability tending to one. A bidirectional version of this algorithm is presented and is shown to dominate the unidirectional one in its average message complexity. Finally, both the unidirectional and the bidirectional algorithms are generalized to perform k selection in the ring, i.e., find the k largest labeled processors.}
+}
+\end{filecontents}
+
+\usepackage{natbib}
+\newtheorem*{lemma}{Lemma}
+\newtheorem*{theorem}{Theorem}
```

Comment:
If you use \newtheorem{theorem}{Theorem}, you can use \begin{lemma} instead of \begin{lemma}[1]